### DISCUSSION TITLE
Should different purposed functions but with same trigger be separated into different tasks (cloud functions)?

### DISCUSSION THREAD

**Discussion by st-trade** [AntlerVC/firetable#769] [Support / Q&A] [answered]
In my project there are two different data transformations that happen when a document in a certain collection is updated. This two transformations have nothing to do with each other, but share the same trigger and triggering conditions.

Would you advise to have to different tasks (cloud functions) with the same triggering and conditions? Or should these be keep in a single tasks?

**Comment by harinij** [comment, marked as answer]
Hi @st-trade Rowy optimizes cloud functions for you and combines various triggering events within a table collection into a single cloud function. This way you have the most optimized cloud functions created, instead of multiple cloud functions getting triggered.

So for example: if you create two task extensions, both triggered on "update" event on a Rowy Table Collection, under the hood they are deployed as one cloud function. Hence we recommend having single tasks for each discreet action, this also makes it easy for you to manage your functions better instead of creating a huge one.

**Reply by st-trade** [reply]
Thanks for the explanation!
